### PR TITLE
[KOGITO-290] - Creating Prometheus annotations from image labels

### PR DESCRIPTION
**Pull request by ricardozanini**
See:
https://issues.jboss.org/browse/KOGITO-290

Take a look at: https://labs.consol.de/development/2018/01/19/openshift_application_monitoring.html

In this PR:
- Added annotations to the Kogito Service pods from `prometheus.io` labels in the image

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by ricardozanini** [comment]
The featuring is working on OCP 3.x, and the pods are getting Prometheus annotations like exemplified below:

[https://ibb.co/nMwfZGQ]()
[https://ibb.co/QHfsS2s]()

The problem is that on OCP 4.x the labels are not being added to the image. I'll work on it this afternoon to figure out this problem.

Meanwhile I'd appreciate a review @xiezhang7 @spolti @sutaakar

**Comment by ricardozanini** [comment]
OCP 4.x bug reported:
https://bugzilla.redhat.com/show_bug.cgi?id=1758305

@sutaakar could you push your friends to take a look? :laughing: 

We had some tests this morning on OCP 3.x and Prometheus is reading the metrics successfully. Thanks to @msmagnanijr that ran those tests.

**Comment by sutaakar** [comment]
:smile:  will see, btw have you tried it on OCP 4.2?

**Comment by ricardozanini** [comment]
> smile will see, btw have you tried it on OCP 4.2?

I don't have access to it. Could you try once this get merged?

**Comment by sutaakar** [comment]
sure

File: pkg/controller/kogitoapp/resource/image_metadata.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+			dc.Spec.Template.Annotations[key] = value
+		}
+	}
+
+	return added
+}
+
+// discoverPortsAndProbesFromImage set Ports and Probes based on labels set on the DockerImage of this DeploymentConfig
+func discoverPortsAndProbesFromImage(dc *appsv1.DeploymentConfig, dockerImage *dockerv10.DockerImage) {
```
Comment (sutaakar) [inline pkg/controller/kogitoapp/resource/image_metadata.go]: Can you please add test coverage for this method, making sure that exposed service label was correctly processed and correct probes were created?

**Comment by sutaakar** [comment]
One comment about test coverage, otherwise good.

File: pkg/controller/kogitoapp/resource/image_metadata.go (hunk 1 above)
Comment (ricardozanini) [reply pkg/controller/kogitoapp/resource/image_metadata.go]: I'll take a look, but I guess we already have. This function is old, we just reallocated in this file.

**Comment by ricardozanini** [comment]
> One comment about test coverage, otherwise good.

That's what you meant?

https://github.com/kiegroup/kogito-cloud-operator/pull/68/files#diff-b954fbaabd70ad3058bd4cd16a4a0c57R64

I guess the probes are not being checked. Will add it.

File: pkg/controller/kogitoapp/resource/image_metadata.go (hunk 1 above)
Comment (sutaakar) [reply pkg/controller/kogitoapp/resource/image_metadata.go]: ok, I just didn't find it in image_metadata_test.go

Comment (ricardozanini) [reply pkg/controller/kogitoapp/resource/image_metadata.go]: I'll add a specific one

**Pull request merged**